src/rcheevos/rc_client.c

```diff
@@ -1939,7 +1939,8 @@ static rc_client_async_handle_t* rc_client_load_game(rc_client_load_state_t* loa
   else {
     RC_CLIENT_LOG_INFO_FORMATTED(client, "Identified game: %u (%s)", load_state->hash->game_id, load_state->hash->hash);
 
-    rc_client_begin_fetch_game_data(load_state);
+    if (!rc_client_begin_fetch_game_data(load_state))
+      return NULL;
```

Comment:
> If you can think of a better way of fixing it, then I'll close the PR

I think a better solution is to simply check to see if the load_state is still active before returning its handle on line 1945:

```
  return (client->state.load == load_state) ? &load_state->async_handle : NULL;
```

I'm working on writing a test to verify.

Reply:
#270

Reply:
SGTM, I'll double check it with PCSX2 tonight.

Reply:
Yup, works fine.